### DISCUSSION TITLE
Route issue while running NiceGUI on EC2

### DISCUSSION THREAD

**Discussion by bugsolarz** [zauberzeug/nicegui#2909] [Q&A] [answered]
I am trying to run NiceGUI project on EC2, it is working fine when I am trying to run it on my local
but when I am trying to replicate the same environment on EC2, getting below error..
not sure what I am doing wrong here.

Note: I am using Authentication example that is provided in the NiceGUI examples on Github repo which is giving this error

The error that I am getting is:

```shell
if request.url.path in Client.page_routes.values() and request.url.path not in unrestricted_page_routes:
AttributeError: type object 'Client' has no attribute 'page_routes'
```

**Comment by rodja** [comment]
Are you sure you have the latest version of NiceGUI installed on the server? `Client.page_routes` was introduced in 1.4.

**Reply by bugsolarz** [reply, marked as answer]
Hey, you are right
The issue is with Python version which is 3.7 so even if I am trying to upgrade NiceGUI or do fresh install of NiceGUI, I am getting 1.12 version which does not have `Client.page_routes` and hence the error. Thank you.
